src/open_company_web/components/finances/finances.cljs

```diff
@@ -72,9 +126,65 @@
   (filter #(not (nil? %))
           (vec (map (fn [[k v]] (clean-data v)) finances-data))))
 
+(defn can-save [owner data]
+  (cond
+    ; new section
+    (om/get-state owner :oc-editing)
+    ; finances has data and title is not empty
+    (and (has-not-zero-data owner data)
+         (> (count (om/get-state owner :title) 0)))
+
+    ; title data and notes editing
+    (and (om/get-state owner :title-editing)
+         (om/get-state owner :data-editing)
+         (om/get-state owner :notes-editing))
+    ; finances has data and at least one btw title data and notes has changes
+    (and (has-not-zero-data owner data)
+         (or (has-title-changes owner data)
+             (has-data-changes owner data)
+             (has-notes-changes owner data)))
+
+    ; title and data editing
+    (and (om/get-state owner :title-editing)
+         (om/get-state owner :data-editing))
+    ; finances has data and title or data has changes
+    (and (has-not-zero-data owner data)
+         (or (has-title-changes owner data)
+             (has-data-changes owner data)))
+
+    ; title and notes editing
+    (and (om/get-state owner :title-editing)
+         (om/get-state owner :notes-editing))
+    ; finances has data and title or data has changes
+    (or (has-title-changes owner data)
+        (has-notes-changes owner data))
+
+    ; data and notes editing
+    (and (om/get-state owner :data-editing)
+         (om/get-state owner :notes-editing))
+    ; finances has data and data or notes has changes
+    (and (has-not-zero-data owner data)
+         (or (has-data-changes owner data)
+             (has-notes-changes owner data)))
+
+    ; title editing
+    (om/get-state owner :title-editing)
+    ; title has changes
+    (has-title-changes owner data)
+
+    ; data editing
+    (om/get-state owner :data-editing)
+    ; finances has data and changes
+    (and (has-data-changes owner data)
+         (has-not-zero-data owner data))
+
+    ; notes editing
+    (om/get-state owner :notes-editing)
+    ; notes has changes
+    (has-notes-changes owner data)))
+
```

Comment:
@belucid i rewrote the save check function. Let me know if you think this needs more improvements or comments